azurelinuxagent/common/cgroups.py

```diff
@@ -0,0 +1,698 @@
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+
+import errno
+import os
+import re
+import time
+
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.utils import fileutil
+from azurelinuxagent.common.version import AGENT_NAME, CURRENT_VERSION
+# from azurelinuxagent.common.protocol.restapi import ExtHandler
+
+BASE_CGROUPS = '/sys/fs/cgroup'
+HIERARCHIES = ['cpu', 'memory']
```

Comment:
Reading the code it was not intuitive what HIERARCHIES meant.  I would suggest METRIC_HIERARCHIES, or something similar.

Reply:
The cgroup documentation calls these "hierarchies", but I take your intent.